### ISSUE TITLE
Find and replace should focus the highlighted result to make it instantly editable

### ISSUE THREAD

**Issue by mabryl** [ckeditor/ckeditor5#13528]
## 📝 Provide a description of the improvement

Right now when Find and replace searches for text it highlights it but doesn't seem to focus/select it. The found text should be selected so that it can be styled, e.g. by pressing the Bold button. The flow would look something like this:

1. A user searches for some text via Find and replace
2. A user selects a specific instance of this text by pressing the up/down arrows in the modal
3. The text can now be styled by pressing Bold, Italic, etc. on the editor's toolbar 

## 📃 Other details

This potential for improvement was brought to our attention by one of our customers via Zendesk.

---

If you'd like to see this improvement implemented, add a 👍 reaction to this post.

**Comment by scofalik** [comment]
I don't agree with this ticket. No software I use works this way. I believe it is against good a11y standards. Finally, we need to keep focus in the dropdown so that keyboard integration can work correctly. I rarely use find feature with mouse.

Instead, we should consider moving selection to the highlighted entry. This way when the editor is re-focused, the selection would be where it should. This is how I work with IDE: ; rather I ctrl+f, write phrase, use enter to get where I want, press esc, make changes.

However, right now, when you press esc, you are still in the toolbar. This is a problem. But since we are moving F&R into a dialog, it will not be a problem anymore.
